src/WorkerChannel.ts

```diff
@@ -407,4 +410,32 @@ export class WorkerChannel implements IWorkerChannel {
             after(context);
         }
     }
+
+    private logColdStartWarning(delayInMs?: number): void {
+        // On reading a js file with function code('require') NodeJs tries to find 'package.json' all the way up to the file system root.
+        // In Azure files it causes a delay during cold start as connection to Azure Files is an expensive operation.
+        if (
+            process.env.WEBSITE_CONTENTAZUREFILECONNECTIONSTRING &&
+            process.env.WEBSITE_CONTENTSHARE &&
+            process.env.AzureWebJobsScriptRoot
+        ) {
+            // Add delay to avoid affecting coldstart
+            if (!delayInMs) {
+                delayInMs = 5000;
+            }
+            setTimeout(() => {
+                const path = require('path');
+                access(path.join(process.env.AzureWebJobsScriptRoot, 'package.json'), constants.F_OK, (e) => {
+                    if (e) {
+                        this.log({
+                            message:
+                                'package.json is not found at the root of the Function App in Azure Files - cold start for NodeJs can be affected.',
+                            level: LogLevel.Debug,

```

Comment:
Should this be `LogLevel.Warning`?

Reply:
Let's leave it as LogLevel. Debug as we do not want to scary customers with sudden unknown warning.